### ISSUE TITLE
Unity 2020.2 (and maybe 2019.3.7+) might need to add receiver tag into LauncherManifest instead of AndroidManifest

### ISSUE THREAD

**Issue by Thaina** [OneSignal/OneSignal-Unity-SDK#312]
**Environment**

* Android
* Unity 2020.2
  * I think this problem occur from 2019.3.7+, ever since unity introduce LauncherManifest build config
* Import firebase messaging and also some other firebase product from UPM

**Description:**

This issue are continued from #310 that was closed

I can't receive push noti from OneSignal since I have upgrade my project to unity 2020.2

After waste my time experimenting since last week. Now I could now send test noti to my android device

The solution for me is copy these lines from onesignal's AndroidManifest.xml and put it in LauncherManifest.xml

```xml
      <receiver
            android:name="com.onesignal.GcmBroadcastReceiver"
            android:permission="com.google.android.c2dm.permission.SEND" >
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
                <category android:name="${manifestApplicationId}" /> <!--replace manifestApplicationId with actual app id-->
            </intent-filter>
        </receiver>
```

Will do some more investigation

**Comment by NilokiloxD** [comment]
Are there any news from the oneSignal guys? The plugin does not work at all in Unity 2020.X, is there any compatibility update planned?

**Comment by rgomezp** [comment]
Howdy,
Thanks @Thaina for the details. We will make sure to update in order to make it compatible.

@NilokiloxD  can you try Thaina's solution?

**Comment by rgomezp** [comment]
Reopening this issue: https://github.com/OneSignal/OneSignal-Unity-SDK/issues/310

**Comment by NilokiloxD** [comment]
There are manual workarounds of course, changing this file alone does only 1 thing. Just saying Unity 2020 needs folders called XX.androidlib to be recognized as Android libs for the manifest or push not icons for example as well as some execution order changes have to be made for the gradle plugin. Cloud build compatibility is also important. Will there be an official update for this or do i have to manually manipulate all these files?

**Comment by rd3-elizeu** [comment]
Any updates on the Unity 2020 compatibility? I cant build an apk. It gives me lots of errors on Lint infrastructure. I can build and run an Android App Bundle to upload to the store and test on my device. But since i can't generate an apk i can't send it to any testers.

**Comment by rd3-elizeu** [comment]
> **Environment**
> 
> * Android
> * Unity 2020.2
>   
>   * I think this problem occur from 2019.3.7+, ever since unity introduce LauncherManifest build config
> * Import firebase messaging and also some other firebase product from UPM
> 
> **Description:**
> 
> This issue are continued from #310 that was closed
> 
> I can't receive push noti from OneSignal since I have upgrade my project to unity 2020.2
> 
> After waste my time experimenting since last week. Now I could now send test noti to my android device
> 
> The solution for me is copy these lines from onesignal's AndroidManifest.xml and put it in LauncherManifest.xml
> 
> ```
>       <receiver
>             android:name="com.onesignal.GcmBroadcastReceiver"
>             android:permission="com.google.android.c2dm.permission.SEND" >
>             <intent-filter>
>                 <action android:name="com.google.android.c2dm.intent.RECEIVE" />
>                 <category android:name="${manifestApplicationId}" /> 
>             </intent-filter>
>         </receiver>
> ```
> 
> Will do some more investigation

I've been trying to export to android studio and generate apk from there. If i activate LauncherManifest and do as you say, my app name changes to com.unity3d.player... and my app icon changes to the default android icon. In this situation i receive notifications. If i don't build using a custom Launcher Manifest my app has the correct name and icon, but even though my device is confirmed as Registered in One Signal website, i  DON't receive any notifications, even if the website states it as Delivered.

**Comment by leszekkrol** [comment]
Aby update in this issue?

**Comment by thatchapitNB** [comment]
I'm use Unity 2020.1.6f.
This my solution is check custom main mainfest, Unity auto create AndroidManifest.xml and copy reciver tag from OneSignalConfig/AndroidManifestTemplate.xml to AndroidManifest.xml.
[https://imgur.com/sBqlBJS](url)

**Comment by OneEon** [comment]
I am getting the following error by doing this: Attribute category#${manifestApplicationId}@name at AndroidManifest.xml:8:17-55 requires a placeholder substitution

**Comment by jkasten2** [comment]
@OneEon You need to replace `${manifestApplicationId}` with your Android package name in your `AndroidManifest.xml`

**Comment by jkasten2** [comment]
A fix for this is now available in [2.13.3 release](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/2.13.3)! You can safely remove the workaround noted here.
